app/controllers/bank_accounts_controller.rb

```diff
@@ -6,9 +6,21 @@ def new
   end
 
   def create
-    @bank_account = BankAccount.new(bank_account_params.merge(student: current_student))
-    unless @bank_account.save
-      render :new
+    if params[:public_token] && params[:account_id] # Creating bank account through Plaid
+      stripe_token = BankAccount.exchange_plaid_token(params[:public_token], params[:account_id])
+      @bank_account = BankAccount.new( {stripe_token: stripe_token, student: current_student, verified: true} )
+      if @bank_account.save
+        @bank_account.ensure_primary_method_exists
+        flash[:notice] = "Your bank account has been confirmed."
+      else
+        flash[:alert] = "There was a problem linking your bank account."
+      end
+      render js: "window.location.pathname ='#{payment_methods_path}'"
+    else # Creating bank account manually
+      @bank_account = BankAccount.new(bank_account_params.merge(student: current_student))
+      unless @bank_account.save
+        render :new
+      end
```

Comment:
I'd like to see most of this logic get pushed down to the model. regardless of whether its plaid or manual, just pass in the params from the request, let the model sort out what to do, and then redirect appropriately based on what type of verification happens next.